mailet/standard/src/main/java/org/apache/james/transport/matchers/AtMost.java

```diff
@@ -33,39 +35,71 @@
 import org.apache.mailet.base.GenericMatcher;
 import org.apache.mailet.base.MailetUtil;
 
+import com.google.common.base.Preconditions;
+import com.google.common.base.Splitter;
 import com.google.common.collect.ImmutableList;
 
 /**
- * Checks that a mail did at most X tries on a specific operation.
+ * <p>Checks that a mail did at most X retries on a specific operation</p>
  *
- * If no retries have been performed previously, it sets up an attribute `AT_MOST_TRIES`
- * in the mail that will be incremented everything the check succeeds.
+ * <p> If no retries have been performed previously for Y attribute, it'll be sets up.</p>
+ * <p> In the mail, every time the check was succeeds, counter of it will be increment by one.
+ * The check fails when the defined X limit is reached.</p>
  *
- * The check fails when the defined X limit is reached.
+ * <ul>
+ * <li>X - count of how many times a specific operation retried</li>
+ * <li>Y - name of attribute represented for specific operation retried, default value is: <i>AT_MOST_TRIES</i></li>
+ * </ul>
  *
- * <p>The example below will match mail with at most 3 tries on the mailet</p>
+ * <p>The example below will match mail with at most 3 tries on the mailet
+ * with attribute name <i>AT_MOST_TRIES</i></p>
  *
  * <pre><code>
- * &lt;mailet match=&quot;AtMost=3&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
+ * &lt;mailet match=&quot;AtMost=AT_MOST_TRIES:3&quot; class=&quot;&lt;any-class&gt;&quot;&gt;
  * &lt;/mailet&gt;
  * </code></pre>
  */
 public class AtMost extends GenericMatcher {
     static final AttributeName AT_MOST_TRIES = AttributeName.of("AT_MOST_TRIES");
-    private Integer atMostRetries;
+    private static final String CONDITION_SEPARATOR = ":";
+    private static final int ONLY_CONDITION_VALUE = 1;
+    private static final int CONDITION_NAME_AND_VALUE = 2;
+
+    private Attribute condition;
 
     @Override
     public void init() throws MessagingException {
-        this.atMostRetries = MailetUtil.getInitParameterAsStrictlyPositiveInteger(getCondition());
+        String conditionConfig = getMatcherConfig().getCondition();
+        Preconditions.checkArgument(StringUtils.isNotBlank(conditionConfig), "MatcherConfiguration is mandatory!");
+        Preconditions.checkArgument(!conditionConfig.startsWith(CONDITION_SEPARATOR),
+            "MatcherConfiguration can not start with '%s'", CONDITION_SEPARATOR);
+
+        List<String> conditions = Splitter.on(CONDITION_SEPARATOR).splitToList(conditionConfig);
+        condition = parseAttribute(conditions);
+    }
+
+    private Attribute parseAttribute(List<String> conditions) throws MessagingException {
```

Comment:
This should return an AttributeName

Reply:
with https://github.com/linagora/james-project/pull/3558#discussion_r451431652, you mean: we should have default value for `attributeName` and `attributeValue` right?

Reply:
If we do so, what about when:
```
<mailet match="AtMost=5" class="<any-class>">
   </mailet>
```
the code will understand '5' is an attributeName and put default value for it, right.I dont think is good for us?

Reply:
You can keep your method parseAttribute I think, but you don't need to return a value. Instead you will initialize two global variables:
* `atMostExecutions` (from previous PR)
* then the attribute name that is optional (that you init at a default value in case missing)

WDYT?